src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -0,0 +1,314 @@
+import { ICredentialSubject, PEX } from '@sphereon/pex';
+import { v4 as uuid } from 'uuid';
+import { SignerService } from '../signer';
+import {
+  verifiableCredentialEIP712Types,
+  verifiablePresentationEIP712Types,
+  verifiablePresentationWithCredentialEIP712Types,
+} from './eip712.types';
+import {
+  VerifiableCredential,
+  VerifiablePresentation,
+  ProofOptions,
+  VerifyVerifiableCredentialResults,
+  CreateCredentialParams,
+  EWFCredentialSubject,
+  Credential,
+  Presentation,
+  CreatePresentationParams,
+} from './verifiable-credentials.types';
+
+export abstract class VerifiableCredentialsServiceBase {
+  /**
+   * @param {string} credential - The credential object in JSON format
+   * @param {string} linked_data_proof_options - The proof options in JSON format
+   * @param {string} public_key - Information about public key that credential will be signed in JSON format
+   * @description get prepared data for signing a credential
+   * @returns {Promise<string>} JSON stringified prepared data (including EIP712 types and credential with proof ready for signing)
+   */
+  protected abstract prepareIssueCredential: (
+    credential: string,
+    linked_data_proof_options: string,
+    public_key: string
+  ) => Promise<string>;
+
+  /**
+   * @param {string} credential - The credential object in JSON format
+   * @param {string} preparation - Output of `prepareIssueCredential` method
+   * @param {string} signature - Signature of the credential
+   * @description get verifiable credential object
+   * @returns {Promise<string>} verifiable credential object in JSON format
+   */
+  protected abstract completeIssueCredential: (
+    credential: string,
+    preparation: string,
+    signature: string
+  ) => Promise<string>;
+
+  /**
+   * @param {string} vc - The verifiable credential object in JSON format
+   * @param {string} proof_options - The proof options in JSON format
+   * @description verify given verifiable credential
+   * @returns {Promise<string>} object with results of verification  in JSON format
+   */
+  protected abstract verifyCredential: (
```

Comment:
I don't know what you mean. Credential and presentation are designed in the same way. The only difference is for VC we can use DIDKit, but for presentation we can't because DIDKit doesn't provide yet a methods for that. So basically 
`prepareIssuePresentation` and ` completeIssuePresentation` can be removed for now, but when the Spruce team resolve two issues that we reported there will be possible to use `prepareIssuePresentation` and ` completeIssuePresentation`.

Reply:
I mean why `verifyCredential:(`, but `verifyPresentation(`

Reply:
Oh I see, thanks for noticing that. Will update